### PR TITLE
Drop support for Symfony v4

### PR DESCRIPTION
**Pull request by keradus**

### CODE REVIEW

**Review by julienfalque — APPROVED** [review]

File: composer.json [hunk 1]
```diff
@@ -21,17 +21,17 @@
...
-        "symfony/filesystem": "^4.4.20 || ^5.0",
-        "symfony/finder": "^4.4.20 || ^5.0",
-        "symfony/options-resolver": "^4.4.20 || ^5.0",
+        "symfony/console": "^5.1.3",
+        "symfony/event-dispatcher": "^5.0",
+        "symfony/filesystem": "^5.0",
+        "symfony/finder": "^5.0",
+        "symfony/options-resolver": "^5.0",
         "symfony/polyfill-mbstring": "^1.23",
         "symfony/polyfill-php72": "^1.23",
```
Comment (derrabus) [inline composer.json:30]: You can drop the PHP 7.2 polyfill.

Comment (derrabus) [reply composer.json:30]: Sure, here you go: #6115

**Review by derrabus — APPROVED** [review]

**Comment by SpacePossum** [comment]
Do we have an idea how many users are "stuck" on SF4 that (still) use the tool as a direct (dev) dependency in their composer.json?

**Comment by keradus** [comment]
very good question, @SpacePossum .
I already run "experimental" releases to collect the numbers
```
3.3.0 - regular release
3.3.1 - PHP 7.2 required
3.3.2 - PHP 7.2 required + Symfony 5 required
```

[so far...](https://packagist.org/packages/friendsofphp/php-cs-fixer/stats#v3.3.0)
```
3.3.0 -   65 installations
3.3.1 -  140 installations
3.3.2 - 2300 installations
```

**Comment by GrahamCampbell** [comment]
I'm not sure dropping this adds much value. We don't have any 4.x specific work arounds in the codebase?

**Comment by keradus** [comment]
on last call, we decided to not include it in newest release that is adding support for PHP 8.1.
I would like to include it in next release.

**Review by localheinz — APPROVED** [review]
👍

**Comment by 94noni** [comment]
Hi, Symfony v4.4 is an LTS and will be [supported](https://symfony.com/releases/4.4) until EO 2022 (2023)
Will this be released for a csfixer v4? or v3.x? thank you

**Comment by GrahamCampbell** [comment]
v3

**Comment by keradus** [comment]
actually released as `v3.5.0`

**Comment by GPHemsley-RELX** [comment]
This seems to me to be a strange decision given that Symfony 4.4 is LTS and there are no Symfony 4.x-specific workarounds in the codebase that would benefit from this - which are both points raised by others above.

I'm also not clear why this was so tightly coupled with the dropping of support for versions of PHP prior to 7.4. Symfony 4.4 supports PHP 7.4.

I am not "stuck" on Symfony 4.4. I have made the deliberate decision to take advantage of the LTS period and not update Symfony yet. But that doesn't mean I'm not updating PHP-CS-Fixer along the way.

**Comment by derrabus** [comment]
> This seems to me to be a strange decision given that Symfony 4.4 is LTS and there are no Symfony 4.x-specific workarounds in the codebase that would benefit from this - which are both points raised by others above.

Before this change, PHP CS Fixer supported a span of three Symfony major versions. That alone complicates the test matrix and makes developing new functionality tedious. It also prevented this project from taking advantage of Symfony 5 features until now. The reason behind Symfony's LTS period is to give everyone enough time to upgrade at their own pace. No downstream open-source project should feel obliged to support an LTS branch until the very last minute.

Given that this project is maintained by a handfull of developers in their freetime, I find this change very reasonable.

> I am not "stuck" on Symfony 4.4. I have made the deliberate decision to take advantage of the LTS period and not update Symfony yet.

Well and this project has made the deliberate decision to abandon Symfony 4.4. A perfectly valid decision which should be respected. PHP CS Fixer supports the current Symfony LTS which is 5.4, see https://symfony.com/releases.

If you insist on staying on Symfony 4.4 (which is totally fine!), these are your options:

* Use PHP CS Fixer 3.4.0, a solid release that will serve you well.
* Install PHP CS Fixer in a separate directory as documented here: https://cs.symfony.com/#installation
* Use the PHAR distribution you can download from https://cs.symfony.com/download/php-cs-fixer-v3.phar
* Use PHIVE to install the PHAR distribution: https://phar.io/

**Comment by keradus** [comment]
@GPHemsley-RELX , would you like to try [shim](https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/5998#issuecomment-1031931620) package?

**Pull request merged**